core/src/main/java/org/infinispan/Version.java

```diff
@@ -24,25 +26,39 @@
 
    private static final Version INSTANCE = new Version();
 
+   private final String version;
+   private final String codename;
    private final byte[] versionId;
    private final String moduleSlot;
    private final short versionShort;
    private final short marshallVersion;
    private final String majorMinor;
    private final String major;
 
+
    private Version() {
-      String parts[] = getParts(Injected.getVersion());
+      Properties properties = new Properties();
+      try (InputStream is = Version.class.getResourceAsStream("/META-INF/infinispan-version.properties")) {
```

Comment:
this approach would expose you to a devilish trick: in case of duplicate Infinispan jars on the classpath you might be logging version Y but executing version X. I'm not against the approach, but it's a risk to be aware of..